TRAMS-API/Repositories/ProjectsRepository.cs

```diff
@@ -0,0 +1,85 @@
+﻿using API.HttpHelpers;
+using API.Models.D365;
+using API.Repositories.Interfaces;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+
+namespace API.Repositories
+{
+    public class ProjectsRepository : IProjectsRepository
+    {
+        private static readonly string _route = "sip_academytransfersprojects";
+
+        private readonly AuthenticatedHttpClient _client;
+        private readonly IOdataUrlBuilder<GetAcademyTransfersProjectsD365Model> _urlBuilder;
+
+        public ProjectsRepository(AuthenticatedHttpClient client,
+                                  IOdataUrlBuilder<GetAcademyTransfersProjectsD365Model> urlBuilder)
+        {
+            _client = client;
+            _urlBuilder = urlBuilder;
+        }
+
+        public async Task<List<GetAcademyTransfersProjectsD365Model>> GetAll()
+        {
+            var url = _urlBuilder.BuildFilterUrl(_route, null);
+
+            await _client.AuthenticateAsync();
+
+            var response = await _client.GetAsync(url);
+            var content = await response.Content?.ReadAsStringAsync();
+
+            if (response.IsSuccessStatusCode)
```

Comment:
Warning: Do we cater for, or log, the error anywhere?  How do we know if the API request to D365 has broken?

Reply:
Not at the moment, will fix in a later PR tackling 429s and other D365 failures